api/src/libs/usersBusinessesMigration.ts

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BusinessesDataClient, UserDataClient } from "@domain/types";
+import { LogWriterType } from "@libs/logWriter";
+import { DynamoDBStreamEvent } from "aws-lambda";
+
+export const usersBusinessesMigration = async (
+  event: DynamoDBStreamEvent,
+  userDataClient: UserDataClient,
+  businessesDataClient: BusinessesDataClient,
+  logger: LogWriterType
+): Promise<void> => {
+  for (const record of event.Records) {
+    logger.LogInfo(`Processing event ${record.eventName} with ID ${record.eventID}`);
+
+    if (
+      (record.eventName === "INSERT" || record.eventName === "MODIFY") &&
+      record.dynamodb &&
+      record.dynamodb.NewImage
+    ) {
+      const userId = record.dynamodb.NewImage.userId.S as string;
+      try {
+        const userData = await userDataClient.get(userId);
+
+        if (!userData || !userData.businesses || Object.keys(userData.businesses).length === 0) {
+          logger.LogError(`No businesses found for user ${userId}. Skipping.`);
+          continue;
+        }
+
+        const businessesData = userData.businesses;
+        for (const businessId in businessesData) {
+          const businessData = businessesData[businessId];
+          const existingBusiness = await businessesDataClient.get(businessId);
+          const currentVersion = existingBusiness ? existingBusiness.version + 1 : 1;
+
+          await businessesDataClient.put({
+            ...businessData,
+            id: businessId,
+            version: currentVersion,
```

Comment:
boulder - I'm not clear on the intent of this field. Everytime I update my data, this fn will get updated, which means the current version will increment. What is the value of this field? How does it let us know which state the data is in?

Reply:
@FarazA22 It seems like you're asking about the intent behind the `version` field and how it helps us track the state of a business. The current implementation increments the version whenever there's a data change for an existing business. For example, if we update the business name or any other property, it increments the business's version. This aligns with our current pattern, where the version is meant to track changes. Regardless of how many times the business is updated via the user's client, the current logic ensures that both the user and business records remain in sync. There is a test in `usersBusinessesMigration.test.ts` that validates this behavior.

I also want to point out that the requirements have shifted slightly. Initially, the logic was designed to set the version to 1 for newly migrated businesses. However, the updated approach ensures that the version is synchronized with the latest `CURRENT_VERSION` from the user object in our codebase. cc @seidior

Reply:
I think it makes sense to create a track changes version, perhaps we should update the variable name so it's more clear.

I'm not sure what you mean by `This aligns with our current pattern, where the version is meant to track changes.` We are tracking UserData version, but that version changes when the shape of the userData is changed, not when a user makes changes.